src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -722,6 +930,11 @@ static int hwmon_sdm_probe(struct platform_device *pdev)
 		sdm->privileged = true;
 	}
 
+	sdm->bdinfo = (struct xocl_sdr_bdinfo*)devm_kzalloc(&sdm->pdev->dev,
```

Comment:
why not embed this data structure into struct xocl_hwmon_sdm so that you don't need to worry about managing it? At least, you need to free it during hwmon_sdm_remove()?

Reply:
IMO, it is better to have separate structure for board sensor data for easy maintenance and easy to track what are all sensors can be received for this BDINFO request.

Now, I replaced kzalloc() with devm_kzalloc(), it is not required to free the memory. It will be freed automatically when drivers are unloaded or when the device is detached from the system.
Thank you for pointing this :)

Reply:
you can have separate data structure, but you don't have to dynamically allocate, right? can you just put the data structure directly into the struct xocl_hwmon_sdm (v.s. putting a pointer over there)?